src/router.zig

```diff
@@ -108,98 +66,176 @@ pub fn Router(comptime Context: type, comptime routes: []const Route(Context)) R
                         .none => return response.notFound(),
                         .static => |index| {
                             inline for (routes) |route, i|
-                                if (index == i) return Self.handle(route, &.{}, context, response, request);
+                                if (index == i) return route.handler(context, response, request, &.{});
                         },
                         .with_params => |object| {
                             inline for (routes) |route, i| {
                                 if (object.data == i)
-                                    return Self.handle(route, object.params[0..object.param_count], context, response, request);
+                                    return route.handler(context, response, request, object.params[0..object.param_count]);
                             }
                         },
                     }
                 },
                 .static => |index| {
                     inline for (routes) |route, i| {
-                        if (index == i) return Self.handle(route, &.{}, context, response, request);
+                        if (index == i) return route.handler(context, response, request, &.{});
                     }
                 },
                 .with_params => |object| {
                     inline for (routes) |route, i| {
                         if (object.data == i)
-                            return Self.handle(route, object.params[0..object.param_count], context, response, request);
+                            return route.handler(context, response, request, object.params[0..object.param_count]);
                     }
                 },
             }
         }
     }.serve;
 }
 
+/// Will automatically convert router captures to function parameter types
+pub fn wrap(comptime Context: type, comptime handler: anytype) Route(Context).Handler {
+    const info = @typeInfo(@TypeOf(handler));
+    if (info != .Fn)
+        @compileError("router.wrap expects a function type");
+
+    const function_info = info.Fn;
+    if (function_info.is_generic)
+        @compileError("Cannot create handler wrapper for generic function");
+    if (function_info.is_var_args)
+        @compileError("Cannot create handler wrapper for variadic function");
+
+    if (function_info.args[0].arg_type.? != Context)
+        @compileError("Argument 0 of a HandlerFn must be Context");
```

Comment:
I think something like this would be nicer:
```zig
@compileError("The first argument must be of type " ++ @typeName(Context) ++ " but found type " ++ @typeName(function_info.args[0].arg_type.?) ++ " instead");
```